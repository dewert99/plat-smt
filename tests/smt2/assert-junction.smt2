(declare-const a Bool)
(declare-const b Bool)
(declare-const c Bool)
(declare-const d Bool)
(push)
(assert (= a (and a b false)))
(assert (= b (and b c (not c))))
(assert (= c (and c d)))
(assert (= d (and)))
(check-sat)
(get-model)
(pop)

(push)
(assert (= a (or a b false)))
(assert (= b (not (or b c (not c)))))
(assert (= c (or c d)))
(assert (= d (and c d)))
(check-sat)
(get-model)
(pop)

(push)
(assert (= a (not (or a b c d))))
(push)
(assert a)
(check-sat)
(pop)
(assert (and (not b) (not c)))
(check-sat)
(get-model)
(assert (not d))
(check-sat)
(pop)

(push)
(assert (= (not a) (and a b c d)))
(push)
(assert (not a))
(check-sat)
(pop)
(assert (and b c))
(check-sat)
(get-model)
(assert d)
(check-sat)
(pop)

