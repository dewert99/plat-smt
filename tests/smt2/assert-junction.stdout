sat
(
 (define-fun a () Bool false)
 (define-fun b () Bool false)
 (define-fun c () Bool false)
 (define-fun d () Bool true)
)
sat
(
 (define-fun a () Bool false)
 (define-fun b () Bool false)
 (define-fun c () Bool false)
 (define-fun d () Bool false)
)
unsat
sat
(
 (define-fun a () Bool false)
 (define-fun b () Bool false)
 (define-fun c () Bool false)
 (define-fun d () Bool true)
)
unsat
unsat
sat
(
 (define-fun a () Bool true)
 (define-fun b () Bool true)
 (define-fun c () Bool true)
 (define-fun d () Bool false)
)
unsat
